Z/Crypto/KDF.hs

```diff
@@ -0,0 +1,25 @@
+module Z.Crypto.KDF where
+
+import Data.Word (Word8)
+import Z.Botan.FFI (hs_botan_kdf)
+import Z.Data.CBytes (CBytes, withCBytesUnsafe)
+import qualified Z.Data.Vector as V
+import Z.Foreign (allocPrimVectorUnsafe, withPrimVectorUnsafe)
+
+kdf ::
+  -- | the name of the given PBKDF algorithm
+  CBytes ->
+  -- | length of output key
+  Int ->
+  -- | passphrase
+  V.Bytes ->
+  -- | salt
+  V.Bytes ->
+  IO Int
```

Comment:
Please check this haddock syntax, for example:

```hs
-- | The description of the function
--
-- return ....
functionA
  :: CBytes
  -- ^ the first argument
  -> IO ()
```

ref: https://haskell-haddock.readthedocs.io/en/latest/markup.html

Reply:
thx